pg_shard.c

```diff
@@ -813,6 +818,46 @@ ExtractFromExpressionWalker(Node *node, List **qualifierList)
 
 
 /*
+ * RemoveFromExpressionQualsMutator walks over a query and removes all quals
+ * from any FromExpr nodes it encounters. Returns a copy of the original query.
+ */
+static Node *
+RemoveFromExpressionQualsMutator(Node *originalNode, void *context)
```

Comment:
This PR then doesn't handle the mutable clauses use-case? It appears to remove all the clauses from local evaluation?